nuxeo-management-rest-api/src/test/java/org/nuxeo/rest/management/TestElasticsearchObject.java

```diff
@@ -0,0 +1,202 @@
+                + "?query=Select * From Document where dc:title like 'Title of my-file%'";
+
+        // Start the ES indexing document that match the nxql query (2 files)
+        try (CloseableClientResponse response = httpClientRule.post(path, null)) {
+            verifyIndexingResponse(response, 2);
+        }
+    }
+
+    @Test
+    public void shouldRunIndexingOnDocumentAndItsChildren() throws IOException {
+        // Init indexes and drop if any
+        esa.initIndexes(true);
+
+        // Create new documents without indexing them
+        createDocuments();
+
+        DocumentModelList documents = coreSession.query("Select * from Document where dc:title = 'My Folder'");
+        String path = String.format("site/management/elasticsearch/%s/%s/reindex", coreSession.getRepositoryName(),
+                documents.get(0).getId());
+
+        // Start the ES indexing for a given document (folder) and its children (2 files)
+        try (CloseableClientResponse response = httpClientRule.post(path, null)) {
+            verifyIndexingResponse(response, 3);
+        }
+    }
+
+    @Test
+    public void shouldFailRunningIndexingWhenRepositoryNotExists() {
+        // Launch the ES indexing
+        String path = String.format(ES_INDEXING_PATH, "unExistingRepository");
+        try (CloseableClientResponse response = httpClientRule.post(path, null)) {
+            assertEquals(SC_NOT_FOUND, response.getStatus());
+        }
+    }
+
+    /**
+     * Allows us to verify the indexing process and its progress status.
+     */
+    protected void verifyIndexingResponse(CloseableClientResponse response, long expectedIndexedDocuments)
+            throws IOException {
+        assertEquals(SC_CREATED, response.getStatus());
+        assertNotNull(response.getLocation());
+
+        // Check the indexing status: at this step the indexing is launched but we are not sure about the exactly
+        // value of its progress status
+        try (CloseableClientResponse clientResponse = httpClientRule.get(response.getLocation().getPath())) {
+            assertEquals(SC_OK, clientResponse.getStatus());
+            JsonNode jsonNode = mapper.readTree(clientResponse.getEntityInputStream());
+            assertFalse(Arrays.asList("UNKNOWN", "ABORTED").contains(jsonNode.get("status").asText()));
+        }
+
+        // Wait until the end of the ES indexing
+        waitForIndexing();
+        assertEquals(expectedIndexedDocuments,
+                ess.query(new NxQueryBuilder(coreSession).nxql(GET_DOCUMENTS_QUERY)).totalSize());
+
+        // Check the indexing status: at this step the indexing should be completed
+        try (CloseableClientResponse clientResponse = httpClientRule.get(response.getLocation().getPath())) {
+            assertEquals(SC_OK, clientResponse.getStatus());
+            JsonNode jsonNode = mapper.readTree(clientResponse.getEntityInputStream());
+            assertEquals("COMPLETED", jsonNode.get("status").asText());
```

Comment:
Do we have other interesting information in the status such as the number of successfully indexed documents?

Reply:
yep, it `protected Long processed;`, i add the check on it + on query result.

Reply:
as discuss the processed is available but on the status endpoint, as decided, a new endpoint will be created on BAF which allow retrieving the processed value. which means until that i don't see a way to control on the total documents treated by BAF :(